client/src/hooks/useApiCallback.js

```diff
@@ -20,7 +20,7 @@ const dataFetchReducer = (state, action) => {
         data: action.payload,
       };
     case types.FETCH_FAILURE:
-      return { ...state, isLoading: false, isError: true };
+      return { ...state, isLoading: false, isError: true, error: action.payload };
```

Comment:
This is super useful!  You need to include the error property in the reducer and the initial values

Reply:
```javascript
const types = {
  FETCH_INIT: 'FETCH_INIT',
  FETCH_SUCCESS: 'FETCH_SUCCESS',
  FETCH_FAILURE: 'FETCH_FAILURE',
};

const dataFetchReducer = (state, action) => {
  switch (action.type) {
    case types.FETCH_INIT:
      return { ...state, isLoading: true, isError: false, error: null };
    case types.FETCH_SUCCESS:
      return {
        ...state,
        isLoading: false,
        isError: false,
        error: null,
        data: action.payload,
      };
    case types.FETCH_FAILURE:
      return { ...state, isLoading: false, isError: true, error: action.payload };
    default:
      throw new Error();
  }
};

const useApiCallback = (method, url, initialData) => {
  const [state, dispatch] = useReducer(dataFetchReducer, {
    isLoading: false,
    isError: false,
    data: initialData,
    error: null,
  });
```

Reply:
@amusameh does this look right?

Reply:
@arrested-developer yeah that looks great to me. will help us render the right error message 👍

Reply:
Happy to add in reducer and into initial values, but what's the reasoning why? Was it throwing an issue for you or is this best practice?

Reply:
Mainly because we might want to make multiple calls with the callback function, and any existing error would need to be cleared from state before making another API call with the same callback function.

If we don't declare the `error` property in the initial state, then we can't destructure it from the state initially returned by the hook.

Reply:
OK cool, thanks for the explanation 👍 Will make the change and push back up